src/table/virtualized-table/HeaderCell.tsx

```diff
@@ -23,9 +23,9 @@ const HeaderCell = ({ index, style, data }: HeaderCellProps) => {
     columResizeHandler,
   } = data;
 
-  const { layout, constraints } = useContextSelector(TableContext, (value) => value.baseProps);
+  const { layout, selectionsAPI, constraints } = useContextSelector(TableContext, (value) => value.baseProps);
   const showRightBorder = useContextSelector(TableContext, (value) => value.showRightBorder);
-  const isInSelectionMode = useContextSelector(TableContext, (value) => value.baseProps.selectionsAPI.isModal());
+  const isInSelectionMode = selectionsAPI?.isModal();
```

Comment:
A recall there being good reason for the "selector" to return the value of isModal. Maybe @haxxmaxx remembers?

Reply:
at least for the pagination table, not exactly sure here. The reason is that we want to trigger a reload when the return of isModal changes (e.g. going in or out of selection mode). But maybe that happens anyway for the virtualized table? we still use staleLayout right?

Reply:
I think it's safest to just leave the selector as it was before with optional chaining to fix the type error.